plenum/test/consensus/order_service/sim_helper.py

```diff
@@ -71,7 +58,23 @@ def check_batch_count(node, expected_pp_seq_no):
     return node._orderer.last_ordered_3pc[1] == expected_pp_seq_no
 
 
+def check_ledger_size(node, expected_ledger_size, ledger_id=DOMAIN_LEDGER_ID):
+    current_ledger_size = node._write_manager.database_manager.get_ledger(ledger_id).size
+    logger.debug("Node: {}, Actual ledger_size: {}, Expected ledger_size is: {}".format(node,
+                                                                                    current_ledger_size,
+                                                                                    expected_ledger_size))
+    return current_ledger_size == expected_ledger_size
+
+
+def get_pools_ledger_size(pool, ledger_id=DOMAIN_LEDGER_ID):
+    sizes = set([node._write_manager.database_manager.get_ledger(ledger_id).size for node in pool.nodes])
+    assert len(sizes) == 1
+    return sizes.pop()
+
+
 def order_requests(pool):
     primary_nodes = [n for n in pool.nodes if n._data.is_primary]
     if primary_nodes and not primary_nodes[0]._data.waiting_for_new_view:
         primary_nodes[0]._orderer.send_3pc_batch()
```

Comment:
It looks like we can call `send_3pc_batch()` for all the nodes without checks like `waiting_for_new_view` or `is_primary` since these checks are done (and supposed to be done) inside `send_3pc_batch`